### PR TITLE
Add a page with a table list of hardware regs

### PR DESCRIPTION
**Pull request by leinacc**
* R/W taken from the regs section name, they may need correcting
* Should wave ram be included (and if so, in this format)?
* Potential CGB/Non-CGB columns?

Fixes https://github.com/gbdev/pandocs/issues/399

### CODE REVIEW

File: src/Hardware_Reg_List.md [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+| $FF69      | [BCPD/BGPD (CGB Mode only)][BCPD/BGPD (CGB Mode only)] | Background color palette data / Background palette data           | R/W |
+| $FF6A      | [OCPS/OBPI (CGB Mode only)][OCPS/OBPI (CGB Mode only)] | OBJ color palette specification / OBJ palette index               | R/W |
+| $FF6B      | [OCPD/OBPD (CGB Mode only)][OCPD/OBPD (CGB Mode only)] | OBJ color palette data / OBJ palette data                         | R/W |
+| $FF6C      | [OPRI (CGB Mode only)][OPRI (CGB Mode only)]           | Object priority mode                                              | R/W |
+| $FF70      | [SVBK (CGB Mode only)][SVBK (CGB Mode only)]           | WRAM bank                                                         | R/W |
+| $FF76      | [PCM12 (CGB Mode only)][PCM12 (CGB Mode only)]         | Digital outputs 1 & 2                                             | R   |
+| $FF77      | [PCM34 (CGB Mode only)][PCM34 (CGB Mode only)]         | Digital outputs 3 & 4                                             | R   |
+| $FFFF      | [IE][IE]                                               | Interrupt enable                                                  | R/W |
```
Comment (ISSOtm) [inline src/Hardware_Reg_List.md:64]:
```suggestion
Address    | Name                        | Description                                                       | Readable / Writable
:----------|-----------------------------|-------------------------------------------------------------------|----
$FF00      | [P1/JOYP]                   | Joypad                                                            | R/W
$FF01      | [SB]                        | Serial transfer data                                              | R/W
$FF02      | [SC]                        | Serial transfer control                                           | R/W
$FF04      | [DIV]                       | Divider register                                                  | R/W
$FF05      | [TIMA]                      | Timer counter                                                     | R/W
$FF06      | [TMA]                       | Timer modulo                                                      | R/W
$FF07      | [TAC]                       | Timer control                                                     | R/W
$FF0F      | [IF]                        | Interrupt flag                                                    | R/W
$FF10      | [NR10]                      | Channel 1 sweep                                                   | R/W
$FF11      | [NR11]                      | Channel 1 length timer & duty cycle                               | R/W
$FF12      | [NR12]                      | Channel 1 volume & envelope                                       | R/W
$FF13      | [NR13]                      | Channel 1 wavelength low                                          | W  
$FF14      | [NR14]                      | Channel 1 wavelength high & control                               | R/W
$FF16      | [NR21]                      | Channel 2 length timer & duty cycle                               | R/W
$FF17      | [NR22]                      | Channel 2 volume & envelope                                       | R/W
$FF18      | [NR23]                      | Channel 2 wavelength low                                          | W  
$FF19      | [NR24]                      | Channel 2 wavelength high & control                               | R/W
$FF1A      | [NR30]                      | Channel 3 DAC enable                                              | R/W
$FF1B      | [NR31]                      | Channel 3 length timer                                            | W  
$FF1C      | [NR32]                      | Channel 3 output level                                            | R/W
$FF1D      | [NR33]                      | Channel 3 wavelength low                                          | W  
$FF1E      | [NR34]                      | Channel 3 wavelength high & control                               | R/W
$FF20      | [NR41]                      | Channel 4 length timer                                            | W  
$FF21      | [NR42]                      | Channel 4 volume & envelope                                       | R/W
$FF22      | [NR43]                      | Channel 4 frequency & randomness                                  | R/W
$FF23      | [NR44]                      | Channel 4 control                                                 | R/W
$FF24      | [NR50]                      | Master volume & VIN panning                                       | R/W
$FF25      | [NR51]                      | Sound panning                                                     | R/W
$FF26      | [NR52]                      | Sound on/off                                                      | R/W
$FF30-FF3F | [Wave pattern ram]          | Wave pattern RAM                                                  | R/W
$FF40      | [LCDC]                      | LCD control                                                       | R/W
$FF41      | [STAT]                      | LCD status                                                        | R/W
$FF42      | [SCY]                       | Viewport Y position                                               | R/W
$FF43      | [SCX]                       | Viewport X position                                               | R/W
$FF44      | [LY]                        | LCD Y coordinate                                                  | R  
$FF45      | [LYC]                       | LY compare                                                        | R/W
$FF46      | [DMA]                       | OAM DMA source address & start                                    | R/W
$FF47      | [BGP (Non-CGB Mode only)]   | BG palette data                                                   | R/W
$FF48      | [OBP0 (Non-CGB Mode only)]  | OBJ palette 0 data                                                | R/W
$FF49      | [OBP1 (Non-CGB Mode only)]  | OBJ palette 1 data                                                | R/W
$FF4A      | [WY]                        | Window Y position                                                 | R/W
$FF4B      | [WX]                        | Window X position plus 7                                          | R/W
$FF4D      | [KEY1 (CGB Mode only)]      | Prepare speed switch                                              | R/W
$FF4F      | [VBK (CGB Mode only)]       | VRAM bank                                                         | R/W
$FF51      | [HDMA1 (CGB Mode only)]     | VRAM DMA source high                                              | W  
$FF52      | [HDMA2 (CGB Mode only)]     | VRAM DMA source low                                               | W  
$FF53      | [HDMA3 (CGB Mode only)]     | VRAM DMA destination high                                         | W  
$FF54      | [HDMA4 (CGB Mode only)]     | VRAM DMA destination low                                          | W  
$FF55      | [HDMA5 (CGB Mode only)]     | VRAM DMA length/mode/start                                        | R/W
$FF56      | [RP (CGB Mode only)]        | Infrared communications port                                      | R/W
$FF68      | [BCPS/BGPI (CGB Mode only)] | Background color palette specification / Background palette index | R/W
$FF69      | [BCPD/BGPD (CGB Mode only)] | Background color palette data / Background palette data           | R/W
$FF6A      | [OCPS/OBPI (CGB Mode only)] | OBJ color palette specification / OBJ palette index               | R/W
$FF6B      | [OCPD/OBPD (CGB Mode only)] | OBJ color palette data / OBJ palette data                         | R/W
$FF6C      | [OPRI (CGB Mode only)]      | Object priority mode                                              | R/W
$FF70      | [SVBK (CGB Mode only)]      | WRAM bank                                                         | R/W
$FF76      | [PCM12 (CGB Mode only)]     | Digital outputs 1 & 2                                             | R  
$FF77      | [PCM34 (CGB Mode only)]     | Digital outputs 3 & 4                                             | R  
$FFFF      | [IE]                        | Interrupt enable                                                  | R/W
```
I trimmed the table width a bit to ease editing a bit.

Why is the "Address" column specifically left-aligned? I'd think right alignment might make more sense, or none at all.

Additionally, I think some of the names make less sense in context (e.g. `PCMxy` are *audio* digital outputs, but that isn't clear when taken outside of the audio chapters), but I want that to be bikeshed separately, as the changes will conflict with the above.

**Review by ISSOtm — CHANGES_REQUESTED** [review]

**Comment by aaaaaa123456789** [comment]
I'd probably add a "console" column listing which consoles have the register in question: for official registers it will always be "all" or "GBC", but for any undocumented registers we discover, it could be anything. Maybe it could be worth it to also list if a register's status (official, undocumented, etc.)?
As for the R/W labels, some registers have mixed R/W perms and the table should show that.

**Comment by ISSOtm** [comment]
> I'd probably add a "console" column listing which consoles have the register in question: for official registers it will always be "all" or "GBC", but for any undocumented registers we discover, it could be anything.

BGP, OBP0, and OBP1 are DMG-only, though. What should the column be titled? "Available on"?

> Maybe it could be worth it to also list if a register's status (official, undocumented, etc.)?

I'm not sure there is much purpose to that; they exist, isn't that what really matters to us?

> As for the R/W labels, some registers have mixed R/W perms and the table should show that.

How should mixed perms be conveyed?

**Comment by aaaaaa123456789** [comment]
> BGP, OBP0, and OBP1 are DMG-only, though. What should the column be titled? "Available on"?

That works, or just "consoles" or "models". Your call.

> I'm not sure there is much purpose to that; they exist, isn't that what really matters to us?

Official registers have known behavior; undocumented/unofficial registers have probably known behavior and might not appear in any other reimplementations of the console (e.g., SGB, Nintendo's VC, etc.). That's the difference.

> How should mixed perms be conveyed?

I'd just write "mixed". It's clear that the purpose of the column is to show R/W perms, after all.

File: src/Hardware_Reg_List.md (hunk 1 above)
Comment (leinacc) [reply src/Hardware_Reg_List.md:64]: left alignment was a mistake. I had it on all of them initially, but tried to remove them

agreed with the PCM. I'll go through them later and see if they make sense outside context

File: src/Hardware_Reg_List.md [hunk 2]
```diff
@@ -0,0 +1,124 @@
+
+# Hardware Registers
+
+Address    | Name               | Description                                                       | Readable / Writable | Models
+:----------|--------------------|-------------------------------------------------------------------|---------------------|-------
```
Comment (ISSOtm) [inline src/Hardware_Reg_List.md:5]:
```suggestion
-----------|--------------------|-------------------------------------------------------------------|---------------------|-------
```

File: src/Hardware_Reg_List.md [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+$FF1E      | [NR34]             | Channel 3 wavelength high & control                               | Mixed               | All
+$FF20      | [NR41]             | Channel 4 length timer                                            | W                   | All
+$FF21      | [NR42]             | Channel 4 volume & envelope                                       | R/W                 | All
+$FF22      | [NR43]             | Channel 4 frequency & randomness                                  | R/W                 | All
+$FF23      | [NR44]             | Channel 4 control                                                 | Mixed               | All
+$FF24      | [NR50]             | Master volume & VIN panning                                       | R/W                 | All
+$FF25      | [NR51]             | Sound panning                                                     | R/W                 | All
+$FF26      | [NR52]             | Sound on/off                                                      | Mixed               | All
+$FF30-FF3F | [Wave pattern ram] | Wave pattern RAM                                                  | R/W                 | All
```
Comment (ISSOtm) [inline src/Hardware_Reg_List.md:35]:
```suggestion
$FF30-FF3F | [Wave RAM] | Storage for one of the sound channels' waveform | R/W                 | All
```
This is merely one possibility, but I just find the term "wave pattern ram" very nondescript and opaque if you don't have a background in the domain.

File: src/Hardware_Reg_List.md [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+$FF1A      | [NR30]             | Channel 3 DAC enable                                              | R/W                 | All
+$FF1B      | [NR31]             | Channel 3 length timer                                            | W                   | All
+$FF1C      | [NR32]             | Channel 3 output level                                            | R/W                 | All
+$FF1D      | [NR33]             | Channel 3 wavelength low                                          | W                   | All
+$FF1E      | [NR34]             | Channel 3 wavelength high & control                               | Mixed               | All
+$FF20      | [NR41]             | Channel 4 length timer                                            | W                   | All
+$FF21      | [NR42]             | Channel 4 volume & envelope                                       | R/W                 | All
+$FF22      | [NR43]             | Channel 4 frequency & randomness                                  | R/W                 | All
+$FF23      | [NR44]             | Channel 4 control                                                 | Mixed               | All
```
Comment (ISSOtm) [inline src/Hardware_Reg_List.md:31]:
```suggestion
$FF10      | [NR10]             | Sound channel 1 sweep                                            | R/W                 | All
$FF11      | [NR11]             | Sound channel 1 length timer & duty cycle                        | Mixed               | All
$FF12      | [NR12]             | Sound channel 1 volume & envelope                                | R/W                 | All
$FF13      | [NR13]             | Sound channel 1 wavelength low                                   | W                   | All
$FF14      | [NR14]             | Sound channel 1 wavelength high & control                        | Mixed               | All
$FF16      | [NR21]             | Sound channel 2 length timer & duty cycle                        | Mixed               | All
$FF17      | [NR22]             | Sound channel 2 volume & envelope                                | R/W                 | All
$FF18      | [NR23]             | Sound channel 2 wavelength low                                   | W                   | All
$FF19      | [NR24]             | Sound channel 2 wavelength high & control                        | Mixed               | All
$FF1A      | [NR30]             | Sound channel 3 DAC enable                                       | R/W                 | All
$FF1B      | [NR31]             | Sound channel 3 length timer                                     | W                   | All
$FF1C      | [NR32]             | Sound channel 3 output level                                     | R/W                 | All
$FF1D      | [NR33]             | Sound channel 3 wavelength low                                   | W                   | All
$FF1E      | [NR34]             | Sound channel 3 wavelength high & control                        | Mixed               | All
$FF20      | [NR41]             | Sound channel 4 length timer                                     | W                   | All
$FF21      | [NR42]             | Sound channel 4 volume & envelope                                | R/W                 | All
$FF22      | [NR43]             | Sound channel 4 frequency & randomness                           | R/W                 | All
$FF23      | [NR44]             | Sound channel 4 control                                          | Mixed               | All
```

**Review by ISSOtm — CHANGES_REQUESTED** [review]
Bikeshed time! :D

**Review by avivace — APPROVED** [review]

**Review by ISSOtm — APPROVED** [review]

**Pull request merged**